uber/templates/groupform.html

```diff
@@ -32,8 +32,15 @@
     {% endif %}
 {% else %}
     <script>
+        var showOrHideCategoriesText = function() {
+            setVisible('#category-other', $(":checkbox[name=categories][value={{ c.OTHER }}]").prop("checked"));
+        };
+
         $(function () {
             $('#bold-field-message').insertBefore($.field('name').parents('.form-group'));
+            $(":checkbox[name=categories][value={{ c.OTHER }}]").parents("label").append('<input id="category-other" class="form-control" type="text" name="categories_text" value="{{ group.categories_text }}" />');
```

Comment:
This dynamic generation of HTML is probably unnecessary. Rendering the HTML inline will work just as well:
```
<div class="form-group">
    <label for="categories" class="col-sm-3 control-label">What kinds of things do you sell?</label>
    <div class="col-sm-9">
        {{ macros.checkgroup(group, 'categories') }}
        <input id="category-other" class="form-control" type="text" name="categories_text" value="{{ group.categories_text }}"/>
    </div>
</div>
```

Although, there are some layout issues with placing the `<input>` element where it is:

<img width="593" alt="screen shot 2017-07-21 at 12 59 25 pm" src="https://user-images.githubusercontent.com/2592431/28474712-9e376718-6e17-11e7-90ef-c8a127b21dde.png">


It works, but it could look better. One option would be to make the `<label class="checkgroup">` elements `display: block`:

<img width="839" alt="screen shot 2017-07-21 at 1 07 19 pm" src="https://user-images.githubusercontent.com/2592431/28474815-f43715fa-6e17-11e7-9db5-a9760a04bb79.png">


My personal preference would be to put the `'<input id="category-other">` in it's own column, with some placeholder text:
```
<div class="form-group">
    <label for="categories" class="col-sm-3 control-label">What kinds of things do you sell?</label>
    <div class="col-sm-9">
        {{ macros.checkgroup(group, 'categories') }}
    </div>
    <div class="col-sm-6 col-sm-offset-3">
        <input id="category-other" class="form-control" type="text" name="categories_text" value="{{ group.categories_text }}" placeholder="Other category"/>
    </div>
</div>
```

Which ends up rendering like this on large screens:

<img width="838" alt="screen shot 2017-07-21 at 1 12 36 pm" src="https://user-images.githubusercontent.com/2592431/28474885-4111d248-6e18-11e7-8871-9ea28f68a34c.png">


And this on small screens:

<img width="578" alt="screen shot 2017-07-21 at 1 13 03 pm" src="https://user-images.githubusercontent.com/2592431/28474911-5ae826d6-6e18-11e7-9357-d86cb710bd77.png">